### PR TITLE
[keymap] submitting ariccb planck keymap

### PR DESCRIPTION
**Pull request by ariccb**
## Description

I had issues with my previous pull request because I had accidentally made commits to the master branch, so I removed that, and re-synced up to the qmk/master using the qmk docs to guide my way. I've got it all settled now and would like to add my keymap to the repo.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/planck/keymaps/ariccb/features/select_word.c [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    const uint8_t all_mods = mods;
+#endif  // NO_ACTION_ONESHOT
+    if ((all_mods & MOD_MASK_SHIFT) == 0) {  // Select word.
+#ifdef MAC_HOTKEYS
+      register_code(KC_LALT);
+#else
+      register_code(KC_LCTL);
+#endif  // MAC_HOTKEYS
+      if (state == STATE_NONE) {
+        SEND_STRING(SS_TAP(X_RGHT) SS_TAP(X_LEFT));
```
Comment (fauxpark) [inline keyboards/planck/keymaps/ariccb/features/select_word.c:46]:
```suggestion
        tap_code(KC_RGHT);
        tap_code(KC_LEFT);
```

File: keyboards/planck/keymaps/ariccb/features/select_word.c [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+      register_code(KC_RGHT);
+      state = STATE_WORD;
+    } else {  // Select line.
+      if (state == STATE_NONE) {
+        clear_mods();
+#ifndef NO_ACTION_ONESHOT
+        clear_oneshot_mods();
+#endif  // NO_ACTION_ONESHOT
+#ifdef MAC_HOTKEYS
+        SEND_STRING(SS_LCTL("a" SS_LSFT("e")));
```
Comment (fauxpark) [inline keyboards/planck/keymaps/ariccb/features/select_word.c:58]:
```suggestion
        register_code16(LCTL(KC_A));
        tap_code16(LSFT(KC_E));
        unregister_code16(LCTL(KC_A));
```

File: keyboards/planck/keymaps/ariccb/features/select_word.c [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+#else
+        SEND_STRING(SS_TAP(X_HOME) SS_LSFT(SS_TAP(X_END)));
```
Comment (fauxpark) [inline keyboards/planck/keymaps/ariccb/features/select_word.c:60]:
```suggestion
        tap_code(KC_HOME);
        tap_code16(LSFT(KC_END));
```

File: keyboards/planck/keymaps/ariccb/config.h [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+   - MIDI notes can be added to the keymap
+   - Octave shift and transpose
+   - Virtual sustain, portamento, and modulation wheel
+   - etc.
+*/
+//#define MIDI_ADVANCED
+
+/* override number of MIDI tone keycodes (each octave adds 12 keycodes and allocates 12 bytes) */
+//#define MIDI_TONE_KEYCODE_OCTAVES 2
+
```
Comment (fauxpark) [inline keyboards/planck/keymaps/ariccb/config.h:44]:
```suggestion
```

File: keyboards/planck/keymaps/ariccb/QMK Configurator JSON/ariccb.json [hunk 5]
```diff
@@ -0,0 +1,361 @@
+{
```
Comment (fauxpark) [inline keyboards/planck/keymaps/ariccb/QMK Configurator JSON/ariccb.json:1]: Please delete this file and folder.

File: keyboards/planck/keymaps/ariccb/Layout.txt [hunk 6]
```diff
@@ -0,0 +1,65 @@
+BASE LAYER
```
Comment (fauxpark) [inline keyboards/planck/keymaps/ariccb/Layout.txt:1]: Please delete this file (or place it in a readme).

File: keyboards/planck/keymaps/ariccb/rules.mk [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+LTO_ENABLE = no 			# Link Time Optimization
+DIP_SWITCH_ENABLE=no  		# Recommended to fix no startup sound issue
+COMBO_ENABLE = yes			# Enable Combos
+
+
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no    	# Breathing sleep LED during USB suspend
+
+
+
```
Comment (fauxpark) [inline keyboards/planck/keymaps/ariccb/rules.mk:28]:
```suggestion
BOOTMAGIC_ENABLE = no
CONSOLE_ENABLE = no
BACKLIGHT_ENABLE = yes
MIDI_ENABLE = yes
TAP_DANCE_ENABLE = yes
DIP_SWITCH_ENABLE = no
COMBO_ENABLE = yes

SRC += features/select_word.c
SRC += features/caps_word.c
```
You only need what you've changed vs. the keyboard-level rules.mk.

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/planck/keymaps/ariccb/Layout.txt (hunk 6 above)
Comment (ariccb) [reply keyboards/planck/keymaps/ariccb/Layout.txt:1]: will do.

**Comment by ariccb** [comment]
I've made the requested changes.

File: keyboards/planck/keymaps/ariccb/readme.md [hunk 8]
```diff
@@ -0,0 +1,76 @@
+QWERTY LAYER
```
Comment (fauxpark) [inline keyboards/planck/keymaps/ariccb/readme.md:1]: Needs to be placed in a code block for the formatting to render correctly.

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/planck/keymaps/ariccb/keymap.c [hunk 9]
```diff
@@ -0,0 +1,675 @@
...
+			rgblight_setrgb (0xFF,  0xFF, 0xFF);
+			break;
+		}
+	  return state;
+}
+
+void persistant_default_layer_set(uint16_t default_layer) {
+  eeconfig_update_default_layer(default_layer);
+  default_layer_set(default_layer);
+}
```
Comment (drashna) [inline keyboards/planck/keymaps/ariccb/keymap.c:310]: Could you remove this function?  

It's been deprecated in favor of the `set_single_persistent_default_layer` function (which is a core function).

https://docs.qmk.fm/#/ref_functions?id=setting-the-persistent-default-layer

File: keyboards/planck/keymaps/ariccb/keymap.c [hunk 10]
```diff
@@ -0,0 +1,675 @@
...
+  const uint8_t mods = get_mods();
+  const uint8_t oneshot_mods = get_oneshot_mods();
+
+  switch (keycode) {
+    case QWERTY:
+      if (record->event.pressed) {
+        #ifdef AUDIO_ENABLE
+          PLAY_SONG(tone_qwerty);
+        #endif
+        persistant_default_layer_set(1UL<<_QWERTY);
```
Comment (drashna) [inline keyboards/planck/keymaps/ariccb/keymap.c:403]:
```suggestion
        set_single_persistent_default_layer(_QWERTY);
```

File: keyboards/planck/keymaps/ariccb/keymap.c [hunk 11]
```diff
@@ -0,0 +1,675 @@
...
+        #endif
+      }
+      return false;
+      break;
+    case COLEMAK:
+      if (record->event.pressed) {
+        #ifdef AUDIO_ENABLE
+          PLAY_SONG(tone_colemak);
+        #endif
+        persistant_default_layer_set(1UL<<_COLEMAK);
```
Comment (drashna) [inline keyboards/planck/keymaps/ariccb/keymap.c:442]:
```suggestion
        set_single_persistent_default_layer(_COLEMAK);
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Comment by ariccb** [comment]
I have made the requested changes, hopefully it's the last review!

**Review by fauxpark — APPROVED** [review]

**Review by drashna — APPROVED** [review]

**Pull request merged**